master/security/icmp-ping.md

```diff
@@ -0,0 +1,134 @@
+---
+title: Use ICMP/ping in policy rules
+---
+
+### Big picture
+
+Use Calico network policy to allow and deny ICMP/ping messages. 
+
+### Value
+
+The **Internet Control Message Protocol (ICMP)** provides valuable network diagnostic functions, but it can also be used maliciously. Attackers can use it to learn about your network, or for DoS attacks. Using Calico network policy, you can control where ICMP is used. For example, you can:
+
+- Allow ICMP ping, but only within your cluster 
+- Allow ICMP for pods launched by operators for diagnostic purposes, but block other uses
+- Temporarily enable ICMP to diagnose a problem, then disable it after the problem is resolved
+- Deny/allow ICMPv4 and/or ICMPv6
+
+### Features
+
+This how-to guide uses the following Calico features:
+
+**GlobalNetworkPolicy** or **NetworkPolicy** with:
+
+- Protocol match for ICMPv4 and ICMPv6
+- icmp/NotICMP match for ICMP type and code
+
+### Concepts
+
+#### ICMP packet type and code
+
+Calico network policy also lets you deny and allow ICMP traffic based on specific types and codes. For example, you can specify ICMP type 5, code 2 to match specific ICMP redirect packets. 
+
+For details, see [ICMP type and code](https://en.wikipedia.org/wiki/Internet_Control_Message_Protocol#Control_messages).
+
+### How to
+
+- [Deny all ICMP](#deny-all-icmp)
+- [Allow ICMP ping only within a cluster](#allow-icmp-ping-only-within-a-cluster)
+- [Allow ICMP matching protocol type and code, only within a cluster](#allow-icmp-matching-protocol-type-and-code-only-within-a-cluster)
+
+#### Deny all ICMP
+
+In this example, we introduce a "deny all ICMP" **GlobalNetworkPolicy**. 
+
+This policy **selects all workloads in the cluster**. It enables a default deny for all workloads, in addition to the explicit ICMP deny rules specified in the policy. For more on default deny policy, see [Calico Network Policy]({{site.baseurl}}/{{page.version}}/security/caliconetworkpolicy).   
+
+If your ultimate goal is to allow some traffic, have your regular "allow" policies in place before applying the global deny-all policy.
```

Comment:
> applying the global deny-all policy

Just a minor nit, but it's not a global deny-all policy. It only denies all ICMP traffic. There's a subtle different between a "deny-all" policy (which explicitly denies all traffic) and the default-deny that occurs when a pod is selected by a policy. I think it's important to keep that consistent throughout the docs, else it becomes even more confusing.

Reply:
Ah, yes. I missed that one reference. Good catch. I actually think we may need to go back to the Calico network policy and clarify this topic more (implicit versus explicit). That was the second How to we wrote so I don't think we had the big picture then.